test/e2e/sriov_smoke_test.go

```diff
@@ -0,0 +1,188 @@
+package e2e
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/vishvananda/netlink"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/k8snetworkplumbingwg/sriov-cni/test/util"
+	net "github.com/k8snetworkplumbingwg/sriov-cni/test/util/net"
+	pod "github.com/k8snetworkplumbingwg/sriov-cni/test/util/pod"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("SR-IOV CNI test", func() {
+	Context("Test with kernel Virtual Function driver", func() {
+		var err error
+		var podObj, podObj2 *corev1.Pod
+		var stdoutString, stderrString, net1Mac string
+		var hostLinksBefore, hostLinksAfter []netlink.Link
+
+		BeforeEach(func() {
+			if testDriverKind != "kernel" {
+				Skip("Tests are not suitable for non kernel drivers.")
+			}
+
+			By("Store host links before tests")
+			hostLinksBefore, err = net.GetHostLinkList()
+			Expect(err).To(BeNil())
+
+			err, netnsErr := net.SetVfsMAC(testPfName, containerNsPath)
+			Expect(err).Should(BeNil())
+			Expect(netnsErr).Should(BeNil())
+		})
+
+		AfterEach(func() {
+			By("Get links after test and move those which were moved from Docker to host, once again to Docker netNs")
+			hostLinksAfter, err = net.GetHostLinkList()
+			Expect(err).To(BeNil())
+
+			err = net.MoveLinksToDocker(containerNsPath, hostLinksBefore, hostLinksAfter)
+			Expect(err).To(BeNil())
+		})
+
+		Context("POD is able to consume SR-IOV interfaces", func() {
+			It("Second interface is available within POD", func() {
+				nadObj := createNAD(*testNs, testNetworkName, testNetworkResName)
+				defer deleteNAD(testNetworkName, nadObj)
+
+				By("Create POD")
+				podObj, err = pod.TryToCreateRunningPod(cs.CoreV1Interface, "test-pod-vf-go", *testNs, testNetworkName, testNetworkResName, util.Timeout)
+				defer deletePod(podObj)
+				Expect(err).To(BeNil())
+
+				By("Verification - Check second network interfaces")
+				stdoutString, stderrString, err = pod.ExecuteCommand(cs.CoreV1Interface, kubeConfig, podObj.Name, *testNs, "test", "ethtool -i eth0")
+				Expect(err).Should(BeNil())
+				Expect(stderrString).Should(Equal(""))
+				Expect(stdoutString).Should(ContainSubstring("driver: veth"))
+
+				stdoutString, stderrString, err = pod.ExecuteCommand(cs.CoreV1Interface, kubeConfig, podObj.Name, *testNs, "test", "ethtool -i net1")
+				Expect(err).Should(BeNil())
+				Expect(stderrString).Should(Equal(""))
+				Expect(stdoutString).Should(ContainSubstring("driver: iavf"))
```

Comment:
nit: should check if its in `supportedKernelDrivers` list from `e2e_test_suite_test.go`